srml/staking/src/lib.rs

```diff
@@ -1212,31 +1158,32 @@ impl<T: Trait> Module<T> {
 		// Increment current era.
 		let current_era = CurrentEra::mutate(|s| { *s += 1; *s });
 
-		// prune journal for last era.
-		<EraSlashJournal<T>>::remove(current_era - 1);
-
 		CurrentEraStartSessionIndex::mutate(|v| {
 			*v = start_session_index;
 		});
 		let bonding_duration = T::BondingDuration::get();
 
-		if current_era > bonding_duration {
-			let first_kept = current_era - bonding_duration;
-			BondedEras::mutate(|bonded| {
-				bonded.push((current_era, start_session_index));
+		BondedEras::mutate(|bonded| {
```

Comment:
nit: the entire funciton seems to be like 

```
/// some high level explanation
function_that_does_it();

/// more logic 
one_liner_that_does_it();
```

i would consider putting this into a `update_bonded_eras() `